### PR TITLE
Week3 homework

### PR DESCRIPTION
**Pull request by sima-milli**

### CODE REVIEW

File: homework-submission/week3/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "week2",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "bootstrap": "^4.1.1",
+    "moment": "^2.22.2",
+    "react": "^16.5.2",
```
Comment (hasan-sh) [inline homework-submission/week3/package.json]: Update to the latest version of React => latest 16.6.0

File: homework-submission/week3/src/App.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import "./App.css";
+import TodoApp from "./todo/TodoApp";
+import todos from "./todo/todos";
+
+class App extends Component {
+  state = {
+    todos
+  };
+
+  handelChange = id => {
```
Comment (hasan-sh) [inline homework-submission/week3/src/App.js]: Very nice usage of the `map` method :100: :1st_place_medal:

File: homework-submission/week3/src/App.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  onAdd = value => {
+    const id = this.state.todos.length + 1;
+    value.id = id;
+    const newTodos = [...this.state.todos, value];
+    this.setState({
+      todos: newTodos
+    });
+  };
+
+  onUpdate = (id, description, deadline) => {
```
Comment (hasan-sh) [inline homework-submission/week3/src/App.js]: Hmm, try to think about whether you're cloning the `this.state.todos`  or changing its values.
It comes to what we've talked about last time: `The Differences Between Primitives&References`. 

So, think about how to savely copy the `this.state.todos` object and THEN change some of its values.

File: homework-submission/week3/src/App.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      const newTodo = {
+        ...todo,
+        done: todo.id === id ? !todo.done : todo.done
+      };
+      return newTodo;
+    });
+    this.setState({ todos: newTodos });
+  };
+
+  onAdd = value => {
```
Comment (hasan-sh) [inline homework-submission/week3/src/App.js]: I'd name the `value` parameter `newTodo`. To make more descriptive.

File: homework-submission/week3/src/todo/TodoInput.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+  onInputChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  onAdd = e => {
+    e.preventDefault();
+    const value = this.state;
```
Comment (hasan-sh) [inline homework-submission/week3/src/todo/TodoInput.js]: Haha are you tired from writing the `this` keyword?! Why don't you use object destructuring? Something like:
`const { description, deadline } = this.state`

File: homework-submission/week3/src/todo/TodoInput.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import React from "react";
+
+class TodoInput extends React.Component {
+  state = {
+    description: "",
+    deadline: "",
+    done: false
+  };
+
+  onInputChange = e => {
```
Comment (hasan-sh) [inline homework-submission/week3/src/todo/TodoInput.js]: Very nice :100:

File: homework-submission/week3/src/todo/TodoItem.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    );
+
+    return (
+      <li>
+        <input type="checkbox" checked={done} onChange={this.props.onChange} />
+        {editAction}
+        <button
+          type="button"
+          className="btn btn-info"
+          style={hideButton(this.state.clicked)}
```
Comment (hasan-sh) [inline homework-submission/week3/src/todo/TodoItem.js]: Nice idea!

I'm wondering what happens if the user doesn't want to edit the todo any more? Can they then dismiss the changes?

File: homework-submission/week3/src/App.js (hunk 3 above)
Comment (sima-milli) [reply homework-submission/week3/src/App.js]: can you provide me with your way to hard copy the `this.state.todos` object ?

Comment (hasan-sh) [reply homework-submission/week3/src/App.js]: Sure, take a look at this: https://devdocs.io/javascript/global_objects/object/assign

_HINT:_ Your `result` variable will be an object, which means you can assign it to a newly created object!!

**Pull request closed without merging**